File: smart-toast/src/main/java/com/coder/zzq/smartshow/toast/PlainToastApi.java
package com.coder.zzq.smartshow.toast;


import androidx.annotation.StringRes;

public interface PlainToastApi {

    // short text toast

    void show(CharSequence msg);

    void show(@StringRes int msg);


    void showAtTop(CharSequence msg);

    void showAtTop(@StringRes int msg);


    void showInCenter(CharSequence msg);

    void showInCenter(@StringRes int msg);


    void showAtLocation(CharSequence msg, int gravity, float xOffsetDp, float yOffsetDp);

    void showAtLocation(@StringRes int msg, int gravity, float xOffsetDp, float yOffsetDp);


    // long text toast

    void showLong(CharSequence msg);

    void showLong(@StringRes int msg);


    void showLongAtTop(CharSequence msg);

    void showLongAtTop(@StringRes int msg);


    void showLongInCenter(CharSequence msg);

    void showLongInCenter(@StringRes int msg);


    void showLongAtLocation(CharSequence msg, int gravity, float xOffsetDp, float yOffsetDp);

    void showLongAtLocation(@StringRes int msg, int gravity, float xOffsetDp, float yOffsetDp);

}

File: smart-toast/src/main/java/com/coder/vincent/smart_toast/PlainToastApi.kt
package com.coder.vincent.smart_toast

import androidx.annotation.StringRes

interface PlainToastApi {
    // short text toast
    fun show(msg: CharSequence)
    fun show(@StringRes msg: Int)
    fun showAtTop(msg: CharSequence)
    fun showAtTop(@StringRes msg: Int)
    fun showInCenter(msg: CharSequence)
    fun showInCenter(@StringRes msg: Int)
    fun showAtLocation(msg: CharSequence, gravity: Int, xOffsetDp: Float, yOffsetDp: Float)
    fun showAtLocation(@StringRes msg: Int, gravity: Int, xOffsetDp: Float, yOffsetDp: Float)

    // long text toast
    fun showLong(msg: CharSequence)
    fun showLong(@StringRes msg: Int)
    fun showLongAtTop(msg: CharSequence)
    fun showLongAtTop(@StringRes msg: Int)
    fun showLongInCenter(msg: CharSequence)
    fun showLongInCenter(@StringRes msg: Int)
    fun showLongAtLocation(msg: CharSequence, gravity: Int, xOffsetDp: Float, yOffsetDp: Float)
    fun showLongAtLocation(@StringRes msg: Int, gravity: Int, xOffsetDp: Float, yOffsetDp: Float)
}